src/unlimited.cpp

```diff
@@ -1162,118 +1162,121 @@ void BuildSeededBloomFilter(CBloomFilter& filterMemPool, std::vector<uint256>& v
+            CTxMemPool::txiter iter;
+            for (uint64_t i = 0; i < vPriority.size(); i++)
             {
-                uint256 childHash = child->GetTx().GetHash();
-                if (!setPriorityMemPoolHashes.count(childHash)) {
-                    setPriorityMemPoolHashes.insert(childHash);
-                    nPrioritySize += child->GetTxSize();
-                    LogPrint("bloom", "add priority child %s with fee %d modified fee %d size %d clearatentry %d priority %f\n", 
+                nPrioritySize += vPriority[i].second->GetTxSize();
+                if (nPrioritySize > nBlockPrioritySize)
+                    break;
+                setPriorityMemPoolHashes.insert(vPriority[i].second->GetTx().GetHash());
+
+                // Add children.  We don't need to look for parents here since they will all be parents.
+                iter = mempool.mapTx.project<0>(vPriority[i].second);
+                BOOST_FOREACH(CTxMemPool::txiter child, mempool.GetMemPoolChildren(iter))
+                {
+                    uint256 childHash = child->GetTx().GetHash();
+                    if (!setPriorityMemPoolHashes.count(childHash)) {
+                        setPriorityMemPoolHashes.insert(childHash);
+                        nPrioritySize += child->GetTxSize();
+                        LogPrint("bloom", "add priority child %s with fee %d modified fee %d size %d clearatentry %d priority %f\n", 
                                        child->GetTx().GetHash().ToString(), child->GetFee(), child->GetModifiedFee(), 
                                        child->GetTxSize(), child->WasClearAtEntry(), child->GetPriority(nHeight));
-                }
-            }            
-        }
-
-        // Create a list of high score transactions. We will multiply this by
-        // a factor of ? to account for any differences between the way Miners include tx's
-        CTxMemPool::indexed_transaction_set::nth_index<3>::type::iterator mi = mempool.mapTx.get<3>().begin();
-        uint64_t nBlockSize = 0;
-        while (mi != mempool.mapTx.get<3>().end())
-        {
-            CTransaction tx = mi->GetTx();
-
-            if (!IsFinalTx(tx, nHeight, nLockTimeCutoff)) {
-                LogPrint("bloom", "tx %s is not final\n", tx.GetHash().ToString());
-                mi++;
-                continue;
+                    }
+                }            
             }
 
-            // If this tx is not accounted for already in the priority set then continue and add
-            // it to the high score set if it can be and also add any parents or children.  Also add
-            // children and parents to the priority set tx's if they have any.
-            iter = mempool.mapTx.project<0>(mi);
-            if (!setHighScoreMemPoolHashes.count(tx.GetHash()))
+            // Create a list of high score transactions. We will multiply this by
+            // a factor of ? to account for any differences between the way Miners include tx's
```

Comment:
I'm confused as to whether the "factor of ?" usage is because the factor is difficult to define here, or whether it's a comment that was meant to be updated with a clearer definition later on...